### ISSUE TITLE
attention_mask size

### ISSUE THREAD

**Issue by xwwwwww** [huggingface/transformers#9071]
# 🚀 Feature request

current attention_mask argument is a tensor of shape [batch_size, sequence_length], 

I'd like it to be a tensor of shape [batch_size, from_seq_length, to_seq_length], as I want to set a different attention mask for a different position.

**Comment by xwwwwww** [comment]
have implemented that by myself.
If needed, I can make a pull request

**Comment by Aolius** [comment]
Hi, I'm also trying to utilize such a customized mask. Would you mind sharing your implementation? Thank you!
